client/my-sites/plan-features-2023-grid/hooks/npm-ready/data-store/use-highlighted-features.ts

```diff
@@ -0,0 +1,42 @@
+import {
+	FEATURE_CUSTOM_DOMAIN,
+	FEATURE_NEWSLETTER_IMPORT_SUBSCRIBERS_FREE,
+	FEATURE_UNLIMITED_SUBSCRIBERS,
+	FEATURE_PAYMENT_TRANSACTION_FEES_10,
+	FEATURE_PAYMENT_TRANSACTION_FEES_8,
+	FEATURE_PAYMENT_TRANSACTION_FEES_4,
+} from '@automattic/calypso-products';
+import type { PlansIntent } from './use-grid-plans';
+
+/**
+ * Returns an array of features that, if they exist for a
+ * particular Plan on a pricing table, will be highlighted
+ * (ie, moved to the top of the list, and bolded).
+ *
+ * If the specified features don't exist for a particular
+ * Plan, they will be ignored.
+ *
+ * The Logic for processing the highlightedFeatures
+ * lives in the adjacent usePlanFeaturesForGridPlans hook.
+ *
+ * TODO clk: move to plans data store
+ */
+const useHighlightedFeatures = (
+	intent: PlansIntent | undefined,
+	isInSignup = false
+): string[] => {
```

Comment:
I think there may be a `FeatureSlug` type somewhere in `@automattic/calypso-products` that we can use for return value.

Reply:
I actually looked for this, but didn't see one. There is a [Feature](https://github.com/Automattic/wp-calypso/blob/0f8ccd518db3beb14ad746df7760b69ef9ed14cb/packages/calypso-products/src/types.ts#L33) type that is just defined as a string. But it is generic and I wasn't sure if this was 100% intended as the same thing/type (it is not used to type the return value of getSlug() [here](https://github.com/Automattic/wp-calypso/blob/0f8ccd518db3beb14ad746df7760b69ef9ed14cb/packages/calypso-products/src/types.ts#L36)).

Reply:
Gotcha. I guess it hasn't evolved yet.

There will never be an end of possible follow-ups with plans code. Pandora's box 😬